libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -448,11 +442,14 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
     private suspend fun handleSessionCanceled(): CompletableDeferred<Boolean> {
         val retVal = CompletableDeferred<Boolean>()
         val cancelEvent = NavigationCancelEvent(PhoneState(context))
+        cancelEvent.arrivalTimestamp = dynamicValues.sessionArrivalTime.get().toString()
         populateNavigationEvent(cancelEvent)
         val result = telemetryEventGate(cancelEvent)
         Log.d(TAG, "CANCEL event sent $result")
         callbackDispatcher.cancelCollectionAndPostFinalEvents().join()
         retVal.complete(true)
+        dynamicValues.sessionStarted.set(false)
+        dynamicValues.sessionArrivalTime.set(null)
```

Comment:
This is not needed it's done in `reset` https://github.com/mapbox/mapbox-navigation-android/blob/9cbe18ce65395b45471ec0f2cbc9ef7d31eea1a9/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L85-L86 https://github.com/mapbox/mapbox-navigation-android/blob/9cbe18ce65395b45471ec0f2cbc9ef7d31eea1a9/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L462-L465 which happens right after `handleSessionCanceled` https://github.com/mapbox/mapbox-navigation-android/blob/9cbe18ce65395b45471ec0f2cbc9ef7d31eea1a9/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L518-L519

Reply:
👍